src/dex-ui-components/presets/CreatePoolForm/CreatePoolForm.tsx

```diff
@@ -0,0 +1,362 @@
+import { FormSettings, useFormSettings } from "../FormSettings";
+import { DefiFormLayout } from "../layouts";
+import { TokenPair } from "../SwapTokensForm/types";
+import { TokenInput } from "../TokenInput";
+import { isEmpty, isNil } from "ramda";
+import { CreatePoolState } from "../../../dex-ui/store/poolsSlice";
+import { AlertDialog, InputSelector, LoadingDialog, SwitchTokenButton } from "../../base";
+import { TransactionStatus } from "../../../dex-ui/store/appSlice";
+import { WarningIcon } from "@chakra-ui/icons";
+import { usePoolsData } from "../../../dex-ui/hooks";
+import { REFRESH_INTERVAL } from "../../../dex-ui/hooks/constants";
+import {
+  getCreatePoolExchangeRate,
+  getTokenBalance,
+  getTokensByUniqueAccountIds,
+  getUniqueTokensFromSelectedOne,
+} from "../utils";
+import { TokenState } from "../types";
+import { CreatePoolTransactionParams } from "./types";
+interface CreatePoolFormProps {
+  isLoading: boolean;
+  pairedAccountBalance: AccountBalanceJson | null;
+  tokenPairs: TokenPair[] | null;
+  transactionState: CreatePoolState;
+  connectionStatus: HashConnectConnectionState;
+  connectToWallet: () => void;
+  sendCreatePoolTransaction: (params: CreatePoolTransactionParams) => Promise<void>;
+  resetCreatePoolState: () => Promise<void>;
+}
+
+export function CreatePoolForm(props: CreatePoolFormProps) {
+  const title = "Create Pool";
+  const createPoolForm = useForm<CreatePoolStateFormData>({
+    defaultValues: {
+      ...InitialCreatePoolFormState,
+    },
+  });
+  const formValues: CreatePoolStateFormData = structuredClone(createPoolForm.getValues());
+
+  const formSettings = useFormSettings({ initialTransactionDeadline: 3.0 });
+
+  createPoolForm.watch([
+    "firstToken.displayAmount",
+    "firstToken.symbol",
+    "secondToken.symbol",
+    "secondToken.displayAmount",
+  ]);
+
+  const [isConfirmCreatePoolDialogOpen, setIsConfirmCreatePoolDialogOpen] = useState(false);
+
+  usePoolsData(REFRESH_INTERVAL);
+
+  const isSubmitButtonDisabled =
+    isEmpty(formValues.firstToken.displayAmount) ||
+    isNil(formValues.firstToken.symbol) ||
+    isEmpty(formValues.secondToken.displayAmount) ||
+    isNil(formValues.secondToken.symbol);
+
+  const successMessage = `Created and added
+                            ${formValues.firstToken.amount.toFixed(6)} 
+                            ${formValues.firstToken.symbol}
+                            and ${formValues.secondToken.amount.toFixed(6)} ${formValues.secondToken.symbol} to pool.`;
+
+  const notification = useNotification({
+    successMessage,
+    transactionState: {
+      transactionWaitingToBeSigned: props.transactionState.status === "in progress",
+      successPayload: props.transactionState.successPayload?.transactionResponse ?? null,
+      errorMessage: props.transactionState.errorMessage,
+    },
+  });
+  const isWalletPaired = props.connectionStatus === HashConnectConnectionState.Paired;
+
+  const isTransactionDeadlineValid =
```

Comment:
These settings variables can be accessed through the `useFormSettings` hook once https://github.com/hashgraph/hedera-accelerator-defi-dex-ui/pull/115 is merged.

Reply:
Done